app/src/main/java/pl/mdanilowski/spotted/repository/CitiesRepository.kt

```diff
@@ -0,0 +1,23 @@
+package pl.mdanilowski.spotted.repository
+
+import io.reactivex.Observable
+import pl.mdanilowski.spotted.data.interactors.DatabaseInteractor
+import pl.mdanilowski.spotted.data.interactors.NetworkInteractor
+import pl.mdanilowski.spotted.data.model.City
+
+interface CitiesRepository {
+    fun getAllCities(): Observable<List<City>>
+}
+
+class CitiesRepositoryImpl(
+    private val databaseInteractor: DatabaseInteractor,
+    private val networkInteractor: NetworkInteractor
+) : CitiesRepository {
+
+    override fun getAllCities(): Observable<List<City>> {
+        val databaseSingle : Observable<List<City>>? = databaseInteractor.getAllCities().toObservable()
+        val networkSingle : Observable<List<City>>? = networkInteractor.getCities().toObservable()
+
+        return Observable.merge(databaseSingle, networkSingle)
```

Comment:
I think this repo patter can work better:
It's a feature from the room - when asking for Single<List<Item>> and there are no elements in db - onError is returned instead of an empty list.
+ with merging, you always ask 2 sources. 
I'd either ask for first element from 2 sources (db, network) or just rely on db, and schedule network updates periodically